lib/Auth/Process/privacyidea.php

```diff
@@ -104,67 +104,77 @@ public function process(&$state)
-		    curl_setopt($curl_instance, CURLOPT_SSL_VERIFYPEER, 0);
-	    }
-	    if(!$response = curl_exec($curl_instance)) {
-	        throw new SimpleSAML_Error_BadRequest("privacyIDEA: Bad request to PI server: " . curl_error($curl_instance));
-        };
-	    SimpleSAML_Logger::debug("privacyIDEA: \n\n\n" . $response . "\n\n\n");
-	    $header_size = curl_getinfo($curl_instance, CURLINFO_HEADER_SIZE);
-	    $body = json_decode(substr($response, $header_size));
-	    try {
-	        $result = $body->result;
-	        $status = $result->status;
-	        $value = $result->value->auth;
-        } catch (Exception $e) {
-	        throw new SimpleSAML_Error_BadRequest( "privacyIDEA: We were not able to read the response from the PI server");
+        if ($transaction_id) {
+            SimpleSAML_Logger::debug("Authenticating with transaction_id: " . $transaction_id);
+            $params["transaction_id"] = $transaction_id;
         }
-
-        if ($status !== True) {
-            throw new SimpleSAML_Error_BadRequest("Valid JSON Response, but some internal error occured in PI server");
-        } else {
-            if ($value !== True){
-                SimpleSAML_Logger::debug("privacyIDEA: Wrong user pass");
-                return False;
-            } else {
-	            SimpleSAML_Logger::debug("privacyIDEA: User authenticated successfully");
-                return True;
-            }
+        if ($signaturedata) {
+            SimpleSAML_Logger::debug("Authenticating with signaturedata: " . urlencode($signaturedata));
+            $params["signaturedata"] = $signaturedata;
+        }
+        if ($clientdata) {
+            SimpleSAML_Logger::debug("Authenticating with clientdata: " . urlencode($clientdata));
+            $params["clientdata"] = $clientdata;
         }
+	    $body = sspmod_privacyidea_Auth_utils::curl($params, null, $cfg, "/validate/samlcheck", true);
+		$attributes = NULL;
 
+	    try {
+		    $result = $body->result;
+		    $status = $result->status;
+		    $value  = $result->value;
+		    $auth   = $value->auth;
+	    } catch (Exception $e) {
+		    throw new SimpleSAML_Error_BadRequest("privacyIDEA: We were not able to read the response from the PI server");
+	    }
 
+	    if ($status !== true) {
+		    throw new SimpleSAML_Error_BadRequest("privacyIDEA: Valid JSON response, but some internal error occured in PI server");
+	    } else {
+		    if ( $auth !== true ) {
+			    SimpleSAML_Logger::debug( "Throwing WRONGUSERPASS" );
+			    $detail = $body->detail;
+			    $message = $detail->message;
+			    if (property_exists($detail, "transaction_id")){
+				    $transaction_id = $detail->transaction_id;
+			    }
+			    if (property_exists( $detail, "attributes")){
+				    $attributes = $detail->attributes;
+				    if (property_exists( $attributes, "u2fSignRequest")){
+					    SimpleSAML_Logger::debug("This is an U2F authentication request");
+					    SimpleSAML_Logger::debug(print_r($attributes, true));
```

Comment:
Currently we do nothing in this branch except logging, that we have a u2f authentication.
I can not remember - is this correct?